prow/plugins/lifecycle/lifecycle_test.go

```diff
@@ -17,133 +17,248 @@ limitations under the License.
 package lifecycle
 
 import (
+	"fmt"
 	"reflect"
+	"sort"
 	"testing"
 	"time"
 
 	"github.com/sirupsen/logrus"
 
 	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
```

Comment:
Why add all this extra weight?

We mostly regret writing this fake and prefer the pattern you're erasing

Reply:
> We mostly regret writing this fake and prefer the pattern you're erasing

Switched to the light weight fake client.

Out of curiosity, why is the heavy fakegithub one bad?

Reply:
Mostly because it makes tests more brittle. Aka if a wide variety of different packages use the fake same client, any change to the client is likely to cause unintended side effects in the other packages' unit tests (or require extra work to ensure they don't break).

For example, someone else might have "improved" the fake to track add/removed labels rather than drop the case. This might have caused problems/additional work for other plugins, now or in the future.

Reply:
@fejta thanks for explaining it!